packages/block-library/src/post-template/edit.js

```diff
@@ -152,20 +152,30 @@ export default function PostTemplateEdit( {
 					per_page: -1,
 					context: 'view',
 				} );
-				// We have to build the tax query for the REST API and use as
-				// keys the taxonomies `rest_base` with the `term ids` as values.
-				const builtTaxQuery = Object.entries( taxQuery ).reduce(
-					( accumulator, [ taxonomySlug, terms ] ) => {
-						const taxonomy = taxonomies?.find(
-							( { slug } ) => slug === taxonomySlug
-						);
-						if ( taxonomy?.rest_base ) {
-							accumulator[ taxonomy?.rest_base ] = terms;
-						}
-						return accumulator;
-					},
-					{}
-				);
+				// Shared utility to build REST API parameters from taxonomy terms.
+				const buildTaxQuery = ( terms, suffix = '' ) => {
+					return Object.entries( terms || {} ).reduce(
+						( accumulator, [ taxonomySlug, termIds ] ) => {
+							const taxonomy = taxonomies?.find(
+								( { slug } ) => slug === taxonomySlug
+							);
+							if ( taxonomy?.rest_base && termIds?.length ) {
+								accumulator[ taxonomy.rest_base + suffix ] =
+									termIds;
+							}
+							return accumulator;
+						},
+						{}
+					);
+				};
+				const builtTaxQuery = buildTaxQuery( taxQuery.include );
+				if ( taxQuery.exclude ) {
```

Comment:
We check if ( taxQuery.exclude ) before building should we also check if ( taxQuery.include ) before buildTaxQuery.